app/controllers/api/v1/offers_controller.rb

```diff
@@ -39,7 +39,7 @@ class OffersController < Api::V1::ApiController
       api :POST, '/v1/offers', "Create an offer"
       param_group :offer
       def create
-        @offer.created_by = current_user
+        @offer.created_by_id = current_user.id unless offer_params.has_key?(:created_by_id)
```

Comment:
@patrixr, What if `created_by_id` comes as nil ? i.e. order_params: {created_by_id: nil} ? 
![Screen Shot 2019-07-03 at 12 04 02 PM](https://user-images.githubusercontent.com/26408955/60568460-88f02d00-9d8a-11e9-9e6d-d75df35cc235.png)

Reply:
You can use 
`unless order_params[:created_by_id].present?`
or maybe just 
`unless order_params[:created_by_id]`

Reply:
I think that's the idea. The admin is allowed to set it to nil. This is for offers that come that have no donor (it really happens!). The specs cover the scenario

Reply:
That's exactly what we want. The goal is to allow staff members to create an order with `created_by_id=nil`

Reply:
ref -> https://jira.crossroads.org.hk/browse/GCW-2679

Reply:
Note that the params will prevent donors from doing that